src/libraries/Native/Unix/System.Native/pal_networkchange.c

```diff
@@ -84,25 +100,59 @@ void SystemNative_ReadEvents(int32_t sock, NetworkChangeEvent onNetworkChange)
     for (struct nlmsghdr* hdr = (struct nlmsghdr*)buffer; NLMSG_OK(hdr, (size_t)len); NLMSG_NEXT(hdr, len))
     {
         switch (hdr->nlmsg_type)
+#elif HAVE_RT_MSGHDR
```

Comment:
This use of ifdefs is making the code very hard to read, at least for me.

Reply:
I agree with @stephentoub. I would suggest to have #if XXX around whole function(s). I think we have precedent fort that  elsewhere - like GetMaxLingerTime()

Reply:
Sure, I will separate them out.
I made it like this because there are overlaps, and more unification can be done in future, e.g.; we can use `read(2)` in case of Linux and reduce the difference (they mainly differ by struct type `nlmsghdr` vs `rt_msghdr`, and for-loop).